route/route.go

```diff
@@ -53,6 +53,12 @@ func New() *Router {
 
 // WithInstrumentation returns a router with instrumentation support.
 func (r *Router) WithInstrumentation(instrh func(handlerName string, handler http.HandlerFunc) http.HandlerFunc) *Router {
+	if r.instrh != nil {
```

Comment:
Why would it be nil?

Reply:
It is the zero value for a function

Reply:
That doesn't answer my question. Where are we calling this with nil, and why?

Reply:
When we call router.New(). This is the existing instrumentation.